docs/root/configuration/http/http_filters/_include/gcp-authn-filter-configuration.yaml

```diff
@@ -0,0 +1,80 @@
+static_resources:
+  listeners:
+  - address:
+      socket_address:
+        address: ::1
+        port_value: 0
+    filter_chains:
+    - filters:
+      - name: "http"
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+          codec_type: HTTP2
+          stat_prefix: "config_test"
+          route_config:
+            name: "route_config_0"
+            virtual_hosts:
+            - name: "integration"
+              domains: ["*"]
+              routes:
+              - match:
+                  prefix: "/"
+                route:
+                  cluster: "cluster_0"
+          http_filters:
+          - name: "envoy.filters.http.gcp_authn"
+            typed_config:
+              "@type": type.googleapis.com/net.envoy.source.extensions.filters.http.metadata.GcpAuthnFilterConfig
+            http_uri:
+              uri: "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/default/identity?audience=[AUDIENCE]"
+              cluster: "gcp_authn"
+              timeout:
+                seconds: 10
+          http_filters:
+          - name: "envoy.filters.http.router"
```

Comment:
this needs a `typed_config` 

there are 2 config validation checks - the first is more like static linting - this one, that has failed, starts a server with the binary+config. It seems the first check is not enforcing everything it should

Reply:
Thanks for the review! PTAL.